pkg/controller/controller_instance.go

```diff
@@ -924,42 +984,47 @@ func (c *controller) pollServiceInstance(serviceClass *v1alpha1.ServiceClass, se
 			return err
 		}
 		toUpdate := clone.(*v1alpha1.ServiceInstance)
-		toUpdate.Status.AsyncOpInProgress = false
-		toUpdate.Status.OperationStartTime = nil
+		c.clearServiceInstanceCurrentOperation(toUpdate)
 
 		// If we were asynchronously deleting a Service Instance, finish
 		// the finalizers.
 		if deleting {
-			err := c.updateServiceInstanceCondition(
+			setServiceInstanceCondition(
 				toUpdate,
 				v1alpha1.ServiceInstanceConditionReady,
 				v1alpha1.ConditionFalse,
 				successDeprovisionReason,
 				successDeprovisionMessage,
 			)
-			if err != nil {
-				return err
-			}
 
 			// Clear the finalizer
 			if finalizers := sets.NewString(toUpdate.Finalizers...); finalizers.Has(v1alpha1.FinalizerServiceCatalog) {
 				finalizers.Delete(v1alpha1.FinalizerServiceCatalog)
-				c.updateServiceInstanceFinalizers(toUpdate, finalizers.List())
+				toUpdate.Finalizers = finalizers.List()
+			}
+
+			if _, err := c.updateServiceInstanceStatus(toUpdate); err != nil {
+				return err
 			}
+
 			c.recorder.Event(instance, api.EventTypeNormal, successDeprovisionReason, successDeprovisionMessage)
+
 			glog.V(5).Infof("Successfully deprovisioned ServiceInstance %v/%v of ServiceClass %v at ServiceBroker %v", instance.Namespace, instance.Name, serviceClass.Name, brokerName)
 		} else {
 			// Create/Update for InstanceCredential has completed successfully,
 			// so set Status.ReconciledGeneration to the Generation used.
 			toUpdate.Status.ReconciledGeneration = toUpdate.Generation
```

Comment:
Can be removed; covered by the `c.clearServiceInstanceCurrentOperation(toUpdate)` at line 987.